### ISSUE TITLE
Missing type definitions for nested Go arrays

### ISSUE THREAD

**Issue by thomas11** [pulumi/pulumi#12629]
This issue is a followup to #9883 which seems fixed. However, in the same Azure `security` namespace, other types are missing:

```
find sdk/pulumi-azure-native-sdk -type d -maxdepth 1 -exec sh -c "cd \"{}\" && go build" \;
# github.com/pulumi/pulumi-azure-native-sdk/security
./application.go:101:16: undefined: ApplicationConditionArrayArrayArrayInput
./pulumiTypes.go:9129:48: undefined: ApplicationConditionArgsArrayArrayInput
./pulumiTypes.go:9168:81: undefined: ApplicationConditionArgsArrayArrayOutput
./pulumiTypes.go:9171:6: undefined: ApplicationConditionArgsArrayArrayOutput
./securityConnectorApplication.go:111:16: undefined: ApplicationConditionArrayArrayArrayInput
```

We only noticed this now since we still had a workaround for #9883 in the Azure native schema generation code.

Schema:

```json
        "azure-native:security:getSecurityConnectorApplication": {
            "description": "Get a specific application for the requested scope by applicationId\nAPI Version: 2022-07-01-preview.",
            "inputs": {
                "properties": {
                    "applicationId": {
                        "type": "string",
                        "description": "The security Application key - unique key for the standard application",
                        "willReplaceOnChanges": true
                    },
                    "resourceGroupName": {
                        "type": "string",
                        "description": "The name of the resource group within the user's subscription. The name is case insensitive.",
                        "willReplaceOnChanges": true
                    },
                    "securityConnectorName": {
                        "type": "string",
                        "description": "The security connector name.",
                        "willReplaceOnChanges": true
                    }
                },
                "type": "object",
                "required": [
                    "applicationId",
                    "resourceGroupName",
                    "securityConnectorName"
                ]
            },
            "outputs": {
                "description": "Security Application over a given scope",
                "properties": {
                    "conditionSets": {
                        "type": "array",
                        "items": {
                            "type": "array",
                            "items": {
                                "type": "array",
                                "items": {
                                    "type": "object",
                                    "$ref": "#/types/azure-native:security:ApplicationConditionResponse"
                                }
                            }
                        },
                        "description": "The application conditionSets - see examples"
                    },
                    "description": {
                        "type": "string",
                        "description": "description of the application"
                    },
                    "displayName": {
                        "type": "string",
                        "description": "display name of the application"
                    },
                    "id": {
                        "type": "string",
                        "description": "Resource Id"
                    },
                    "name": {
                        "type": "string",
                        "description": "Resource name"
                    },
                    "sourceResourceType": {
                        "type": "string",
                        "description": "The application source, what it affects, e.g. Assessments"
                    },
                    "type": {
                        "type": "string",
                        "description": "Resource type"
                    }
                },
                "type": "object",
                "required": [
                    "conditionSets",
                    "id",
                    "name",
                    "sourceResourceType",
                    "type"
                ]
            }
        },
```

FYI @danielrbradley

**Comment by thomas11** [comment]
Note that we're not blocked by this since the previous workaround of omitting this property still works, it just causes an incomplete SDK.
